cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/s3/S3StorageClient.java

```diff
@@ -0,0 +1,57 @@
+package com.linkedin.openhouse.cluster.storage.s3;
+
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import org.apache.iceberg.aws.AwsClientFactory;
+import org.apache.iceberg.aws.S3FileIOAwsClientFactories;
+import org.apache.iceberg.aws.s3.S3FileIOAwsClientFactory;
+import org.apache.iceberg.aws.s3.S3FileIOProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import software.amazon.awssdk.services.s3.S3Client;
+
+public class S3StorageClient implements StorageClient<S3Client> {
+  private static final StorageType.Type S3_TYPE = StorageType.S3;
+
+  @Autowired private StorageProperties storageProperties;
+
+  private S3Client s3;
+
+  @PostConstruct
+  public synchronized void init() {
+    // TODO: Validate properties.
```

Comment:
Can we add validations here and possible move such common logic to `BaseStorageClient`?

Reply:
I believe S3 will have its own validations. What is the expectation for s3/aws access token here? We cannot keep it in plain-text in cluster spec, it would need to be injected into memory out of band.

Reply:
Let me take that back, we could securely rebuild cluster spec in the init container. For HDFS we fetch the delegation token in the init container and write them to a fixed location, so something similar to that. Caspian will need similar functionality but we can solve it when we get there.

Reply:
@HotSushi IS it ok to address this in the next PR? I have added a TODO.

Reply:
Yes SG!